subsys/fs/fat_fs.c

```diff
@@ -458,6 +458,27 @@ static int fatfs_unmount(struct fs_mount_t *mountp)
 	return translate_error(res);
 }
 
+#if defined(CONFIG_FILE_SYSTEM_MKFS)
+
+static int fatfs_mkfs(uintptr_t dev_id, int flags)
```

Comment:
One more suggestion: remove flags and use void * pointer to allow passing backend dependent init object, and providing default behaviour when the pointer to init object is null.
I am using the fatfs_mkfs as an example here, but I mean the change should be general.
`inf flags` gives you nothing and does not matter because you are directly asking subsystem to format a device for you.
For example when `fatfs_mkfs(dev, mkfs_opt)` gets called with NULL for mkfs_opt (the init object), then it will initialize the device as the code below does. But if user does the extra mile and passes own MKFS_PARM type object, the mkfs will format the device as requested.

Reply:
Similar thing should be possible for the LittleFS